### PR TITLE
#744: Common court search criteria checkbox - take into account account dependent court judgments

### PR DESCRIPTION
**Pull request by lukdumi**

### CODE REVIEW

File: saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/infoFormSection.js [hunk 1]
```diff
@@ -191,7 +194,16 @@ function InfoFormSection(options) {
                  
                  if ($optionSelected.index() > 0) {
                      
-                     html += addPhrase($optionSelected.text(), fieldDescription, comma);
+                     var optionSelectedText = $optionSelected.text();
+                     
+                     if ($this.attr("id") === "select-common-court") {
+                         if ($("#ccIncludeDependentCourtJudgments").is(":checked")) {
+                             optionSelectedText = optionSelectedText + $("#ccIncludeDependentCourtJudgments").attr("data-info-section-custom-text");
```
Comment (pawelczak) [inline saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/infoFormSection.js]: Here you need to also add space (**+ " "**):
`optionSelectedText = optionSelectedText + " " + $("#ccIncludeDependentCourtJudgments").attr("data-info-section-custom-text");`

Comment (lukdumi) [inline saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/infoFormSection.js]: :+1:

File: saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/search/courtCriteriaForm.js [hunk 2]
```diff
@@ -90,7 +90,8 @@ var CourtCriteriaForm = (function() {
             //inits common court select with options
             addOptionsToSelect({
                 selectId: "#select-common-court",
-                url: contextPath + "/cc/courts/list"
+                url: contextPath + "/cc/courts/list",
+                createOption: function(court) {return createCcCourtOption(court);}
```
Comment (pawelczak) [inline saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/search/courtCriteriaForm.js]: Here you can pass just function name:
`createOption: createCcCourtOption`

File: saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/search/courtCriteriaForm.js [hunk 3]
```diff
@@ -106,16 +107,77 @@ var CourtCriteriaForm = (function() {
...
+                     .fail(function() {});
+                } else {
+                    $commonCourtDivisionSelect.prop("disabled", "disabled").find("option:gt(0)").remove();
+                    $ccIncludeDependentCourtJudgmentsCheckbox.prop("checked", false);
+                    $ccIncludeDependentCourtJudgmentsCheckbox.attr("disabled", "disabled");
+                }
+            });
+            
+        
+        createDivisionOption = function(id, name) {
```
Comment (pawelczak) [inline saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/search/courtCriteriaForm.js]: Missing `var`

File: saos-webapp/src/test/java/pl/edu/icm/saos/webapp/court/TestCourtsFactory.java [hunk 4]
```diff
@@ -17,15 +17,26 @@
 	
 	//------------------------ LOGIC --------------------------
 	
-	public List<SimpleEntity> getSimpleEntities() {
-		SimpleEntity divisionOne = new SimpleEntity();
+	public List<SimpleEntity> createSimpleEntities() {
+		SimpleEntity entityOne = new SimpleEntity();
 		SimpleEntity divisionTwo = new SimpleEntity();
```
Comment (pawelczak) [inline saos-webapp/src/test/java/pl/edu/icm/saos/webapp/court/TestCourtsFactory.java]: I looks strange when one object is named entityOne and the other one divisionTwo.

File: saos-webapp/src/test/java/pl/edu/icm/saos/webapp/court/TestCourtsFactory.java [hunk 5]
```diff
@@ -17,15 +17,26 @@
...
 		divisionTwo.setName("Wydział 23 Karny");
 		
-		return Lists.newArrayList(divisionOne, divisionTwo);
+		return Lists.newArrayList(entityOne, divisionTwo);
 	}
 
+	
+	public List<SimpleCommonCourt> createSimpleCommonCourts() {
+        SimpleCommonCourt entityOne = new SimpleCommonCourt();
+        SimpleCommonCourt divisionTwo = new SimpleCommonCourt();
```
Comment (pawelczak) [inline saos-webapp/src/test/java/pl/edu/icm/saos/webapp/court/TestCourtsFactory.java]: Same here.

Comment (lukdumi) [inline saos-webapp/src/test/java/pl/edu/icm/saos/webapp/court/TestCourtsFactory.java]: ok

File: saos-webapp/src/test/java/pl/edu/icm/saos/webapp/court/TestCourtsFactory.java (hunk 4 above)
Comment (lukdumi) [inline saos-webapp/src/test/java/pl/edu/icm/saos/webapp/court/TestCourtsFactory.java]: ok

File: saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/search/courtCriteriaForm.js (hunk 2 above)
Comment (lukdumi) [inline saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/search/courtCriteriaForm.js]: :+1:

File: saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/search/courtCriteriaForm.js (hunk 3 above)
Comment (lukdumi) [inline saos-webapp/src/main/webapp/WEB-INF/static/javascript/modules/search/courtCriteriaForm.js]: :+1:

**Pull request closed without merging**